### PR TITLE
Sanity check the type of id2label and label2id arguments of from_pretrained for TokenClassification models

### LINKED ISSUE #20773: trainer.save_model   load error
reference code : https://github.com/bhadreshpsavani/UnderstandingNLP/blob/master/go_emotion_of_transformers_multilabel_text_classification_v2.ipynb

----
----> 6 save_model ='./model/emotion'
----> 7 trainer.save_model(save_model)
----> 8 load_model = AutoModel.from_pretrained(save_model)

4 frames
[/usr/local/lib/python3.8/dist-packages/transformers/configuration_utils.py](https://localhost:8080/#) in __init__(self, **kwargs)
    315                     f"{self.id2label}. The number of labels wil be overwritten to {self.num_labels}."
    316                 )
--> 317             print(id2label)
    318             self.id2label = dict((int(key), value) for key, value in self.id2label.items())
    319             # Keys are always strings in JSON so convert ids to int here.

AttributeError: 'list' object has no attribute 'items'

---

After fine-tuning the emotion classification model with 28 classes, I learned and saved it through trainer() , but the model is not loaded. can someone help?

### PR DESCRIPTION
**Pull request by raghavanone**
Fixes #20773 

Sanity check the type of id2label and label2id arguments of from_pretrained for TokenClassification models

### CODE REVIEW

File: src/transformers/configuration_utils.py [hunk 1]
```diff
@@ -313,7 +313,11 @@ def __init__(self, **kwargs):
         self.finetuning_task = kwargs.pop("finetuning_task", None)
         self.id2label = kwargs.pop("id2label", None)
         self.label2id = kwargs.pop("label2id", None)
+        if self.label2id is not None and not isinstance(self.label2id, dict):
+            raise TypeError("Argument label2id should be of type dict")
```
Comment (sgugger) [inline src/transformers/configuration_utils.py:317]:
```suggestion
            raise ValueError("Argument label2id should be a dictionary.")
```

File: src/transformers/configuration_utils.py [hunk 2]
```diff
@@ -313,7 +313,11 @@ def __init__(self, **kwargs):
...
         if self.id2label is not None:
+            if not isinstance(self.id2label, dict):
+                raise TypeError("Argument id2label should be of type dict")
```
Comment (sgugger) [inline src/transformers/configuration_utils.py:320]:
```suggestion
            if not isinstance(self.id2label, dict):
                raise ValueError("Argument id2label should be a dictionary.")
```

**Review by sgugger — COMMENTED** [review]
Thanks for your PR! I left a few suggestions.

**Comment by raghavanone** [comment]
@sgugger One of TF test is failing. Not sure if it is related to changes .

**Review by sgugger — APPROVED** [review]

**Comment by sgugger** [comment]
No, this isn't linked to the PR. Thanks for your contribution!

**Pull request merged**